processor/src/main/java/org/mapstruct/ap/internal/model/IterableMappingMethod.java

```diff
@@ -379,6 +385,15 @@ else if ( other.selectionParameters != null ) {
             return false;
         }
 
+        if ( this.factoryMethod != null ) {
```

Comment:
I just tried with my IDE, it didn't use `Objects.equals`. I also realized that `Objects` is since Java 7, so we can't use it 😄. Therefore it is OK as it is